### ISSUE TITLE
Assuming localhost for Redis connection

### ISSUE THREAD

**Issue by kgbsmurf** [resque/resque#1651]
https://github.com/resque/resque/blob/1d85f7c705ee8a609531c29035f8cf7cc6d67b19/lib/resque.rb#L141

```
  def redis
    return @data_store if @data_store
    self.redis = Redis.respond_to?(:connect) ? Redis.connect : "localhost:6379"
    self.redis
  end
```

should be something like this:
```
        def redis
            return @data_store if @data_store
            self.redis = Redis.new(url: ENV["REDIS_URL"] || "redis://localhost:6379")
            self.redis
        end
```

Additionally, Redis.connect is depricated and shouldn't be used.
source:  https://www.rubydoc.info/github/redis/redis-rb/Redis.connect

Thanks!

**Comment by rafaelfranca** [comment]
I agree we should not use connect anymore but I don't think `ENV["REDIS_URL"]` should be in resque. If you application wants to use REDIS_URL you can set `Resque.redis=` in your application.

**Comment by rafaelfranca** [comment]
I prefer to not support environment variables. It is possible to change the redis connection already and if people wants to use environment variables they can use that.
